File: techvault-client/src/types/category.ts
/** Matches TechVault.API `CategoryListItemDto` list response. */
export type CategoryListItem = {
  id: number
  name: string
  slug: string
  description?: string | null
  imageUrl?: string | null
  parentCategoryId?: number | null
  displayOrder: number
  isActive: boolean
  productCount: number
}

/** Matches TechVault.API `CategoryDetailDto` (GET `/categories/{slug}`). */
export type CategoryDetail = {
  id: number
  name: string
  slug: string
  description?: string | null
  imageUrl?: string | null
  parentCategoryId?: number | null
  displayOrder: number
  isActive: boolean
  productCount: number
}

export type CreateCategoryBody = {
  name: string
  slug?: string | null
  description?: string | null
  imageUrl?: string | null
  parentCategoryId?: number | null
  displayOrder: number
  isActive: boolean
}

export type UpdateCategoryBody = CreateCategoryBody

File: TechVault.API/Models/Category.cs
namespace TechVault.API.Models;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Slug { get; set; } = null!;
    public string? Description { get; set; }

    /// <summary>Optional hero image URL (relative to site root, e.g. <c>/images/categories/...</c>).</summary>
    public string? ImageUrl { get; set; }
    public int? ParentCategoryId { get; set; }
    public int DisplayOrder { get; set; }
    public bool IsActive { get; set; } = true;

    public Category? ParentCategory { get; set; }
    public ICollection<Category> ChildCategories { get; set; } = new List<Category>();
    public ICollection<Product> Products { get; set; } = new List<Product>();
}
